### ISSUE TITLE
Intellisense stopped working after update

### ISSUE THREAD

**Issue by krzikalla** [microsoft/vscode-cpptools#3878]
**Type: LanguageService**

**Describe the bug**
- OS and Version: SUSE Linux Enterprise Desktop 12 SP2
- VS Code Version: 1.36
- C/C++ Extension Version: 0.24.0
- A clear and concise description of what the bug is.
After updating VS Code from 1.26 and the C/C++ Extension some symbols can't be found anymore.
I am unable to pinpoint the nature of symbols. Apparently it only affects symbols related to templates. Some templates are found though.

**Comment by bobbrow** [comment]
Can you provide a little more information about the problem and maybe a small sample where it doesn't work?  We did integrate the latest set of updates for IntelliSense from Visual Studio, so it's possible that something regressed, but this isn't enough information for us to work with.

**Comment by krzikalla** [comment]
I couldn't create a small sample. The affected project is large and contains a lot of templates.  
It appears to happen always inside class definitions and often also in namespace scope. Toggling Intellisense options seems of no help. Neither switching to Tag Parser nor taking clang-x64 instead of msvc-x64 makes a difference. No problems are shown in the "problems" tab.
(there is an error on startup: "/lib64/libdbus-1.so.3: no version information available" but I guess, it is unrelated).

**Comment by sean-mcmanus** [comment]
Can you give more info about what "symbols can't be found" means? Does the symbol not appear in the Outline view after making an edit to the file? (we have a known issue where symbols won't appear in the Outline view before an edit is made). Is hover or go to definition failing to find the symbol? Is the symbol not found even in the file that defines it? Does it give an error squiggle about something being undefined?

**Comment by krzikalla** [comment]
In the outline view appears nothing - even after editing. Hover and "go to definition" finds very few symbols - and apparently only after editing. The kind of found symbols does not depend on the file location. There are no error squiggles anywhere - not even at the includes (which can't be opened too).

If I have compiler errors in the problem tab, then I can't open them and get the pop-up (Error: File not found (PATH)). PATH is correct though. I am not sure if this is related, since the "cannot-navigate-to-compiler-errors-bug" is actually an old one which I actually hoped to get fixed meanwhile (that was actually the sole reason for me updating to 1.36).

**Comment by sean-mcmanus** [comment]
Do you know what version the C/C++ extension you were using previously? Are you able to use Install Another Version... in the Extension view to use 0.23.1 or an earlier version that works?

If the outline view gives nothing after editing, then there is some problem parsing symbols in the file.

It sounds like your project may not be configured correctly. You could use the C/C++: Log Diagnostics command to see what IntelliSense settings are being used.

**Comment by krzikalla** [comment]
Thanks for the Log Diagnostics  hint. I found error = 0x8064000a, which quickly lead me to #2169. I killed the dangling processes by hand, now it works. 
Nevertheless the error navigation issue is still there.

**Comment by sean-mcmanus** [comment]
What is the error navigation issue? Can you give more repro info on that?

**Comment by sean-mcmanus** [comment]
It sounds like you need to set your problemMatcher to use an absolute path, such as with
```
"problemMatcher": {
"base": "$gcc",
"fileLocation": "absolute"
}
```
we might change the gcc problem matcher use the "autoDetect" in our next release.

**Comment by elahehrashedi** [comment]
This issue is fixed in 0.28.0.
0.28.0 release: https://github.com/microsoft/vscode-cpptools/releases/tag/0.28.0
